### ISSUE TITLE
Enchant fails to find aspellWithDicts dictionaries

### ISSUE THREAD

**Issue by 01mf02** [NixOS/nixpkgs#29429]
## Issue description

Enchant cannot find any aspell dictionaries when aspellWithDicts was used to install aspell. The `aspell` command on its own works fine.

### Steps to reproduce

* Put the following into your `systemPackages`:
  ~~~
  enchant
  (aspellWithDicts (ps : with ps; [ cs de en fr it ]))
  ~~~
* `echo "Hello World" > test.txt`
* `aspell -c test.txt` succeeds, i.e. does not output any errors
* `enchant test.txt` outputs: "Couldn't create a dictionary for en_US.UTF-8"; however, I expected it to succeed just like aspell

## Technical details

* System: NixOS: 17.09beta581.71fb65709c (Hummingbird)
* Nix version: 1.11.14
* Nixpkgs version: 17.09beta581.71fb65709c
* Sandboxing enabled: false

**Comment by 01mf02** [comment]
I looked at the aspell wrapper:

~~~
[michi@nixos:~]$ cat `which aspell`
#! /nix/store/li1r4j6526vy84f8qg1ydfryp55xxxh1-bash-4.4-p12/bin/bash -e
export ASPELL_CONF="dict-dir /nix/store/kzjdfd015f6sf4bxlbvf229013lznchq-aspell-dicts/lib/aspell"
exec "/nix/store/qhb8mq3cl1r6w5pvsr8p9zawkwl538bn-aspell-0.60.6.1/bin/aspell"  "${extraFlagsArray[@]}" "$@"
~~~

With the system-wide export of ASPELL_CONF, enchant works fine:

~~~
export ASPELL_CONF="dict-dir /nix/store/kzjdfd015f6sf4bxlbvf229013lznchq-aspell-dicts/lib/aspell"
enchant test.txt
~~~

**Comment by FRidh** [comment]
Could you check #29465?

**Comment by 01mf02** [comment]
I'd love to help you, but I have just started using NixOS two days ago and do not yet know how to integrate your proposed change into my system.
I tried to add the following to my configuration.nix:

~~~
  nixpkgs.config.packageOverrides = pkgs : {
    aspellWithDicts = (import ./aspell-with-dicts.nix);
  };
~~~

But `nixos-rebuild switch` gave me:

    error: value is a function while a set was expected, at /etc/nixos/configuration.nix:64:6                           

Line 64 is:

~~~
  environment.systemPackages = with pkgs; [
# ...
# now comes line 64:
    (aspellWithDicts (ps : with ps; [ cs de en fr it ]))
# ...
]
~~~

If you could show me how to do this most efficiently (possibly without needing to clone nixpkgs or rebuilding my whole system ^^), I would be most grateful! :)

**Comment by FRidh** [comment]
@01mf02 you can pass `nixos-rebuild switch -I nixpkgs=https://github.com/FRidh/nixpkgs/archive/aspell.tar.gz)` with your original config; no need to import.

**Comment by 01mf02** [comment]
Thank you for your assistance, this worked perfectly. I have commented on #29465.
Out of curiosity, is it possible to integrate single packages like I wanted to do with `import` in `configuration.nix`?

**Comment by FRidh** [comment]
> Out of curiosity, is it possible to integrate single packages like I wanted to do with import in configuration.nix?

You can indeed import Nix expressions, however, if you want to import packages from say another Nixpkgs version/tree, then you best import that Nixpkgs, and then the correct attribute from that set.

Fixed with #29465 so closing.
